Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/organizer/create/TraceLocationCreateFragment.kt

```diff
@@ -0,0 +1,162 @@
+package de.rki.coronawarnapp.ui.eventregistration.organizer.create
+
+import android.os.Bundle
+import android.text.format.DateFormat.is24HourFormat
+import android.view.View
+import androidx.core.widget.doOnTextChanged
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.navArgs
+import com.google.android.material.datepicker.CalendarConstraints
+import com.google.android.material.datepicker.DateValidatorPointForward
+import com.google.android.material.datepicker.MaterialDatePicker
+import com.google.android.material.timepicker.MaterialTimePicker
+import com.google.android.material.timepicker.TimeFormat
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.contactdiary.util.getLocale
+import de.rki.coronawarnapp.contactdiary.util.hideKeyboard
+import de.rki.coronawarnapp.databinding.TraceLocationCreateFragmentBinding
+import de.rki.coronawarnapp.ui.durationpicker.DurationPicker
+import de.rki.coronawarnapp.ui.durationpicker.toContactDiaryFormat
+import de.rki.coronawarnapp.util.di.AutoInject
+import de.rki.coronawarnapp.util.ui.popBackStack
+import de.rki.coronawarnapp.util.ui.viewBindingLazy
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModelsAssisted
+import org.joda.time.DateTimeZone
+import org.joda.time.LocalDate
+import org.joda.time.LocalDateTime
+import org.joda.time.LocalTime
+import javax.inject.Inject
+
+class TraceLocationCreateFragment : Fragment(R.layout.trace_location_create_fragment), AutoInject {
+
+    private val binding: TraceLocationCreateFragmentBinding by viewBindingLazy()
+    private val navArgs by navArgs<TraceLocationCreateFragmentArgs>()
+
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+    private val viewModel: TraceLocationCreateViewModel by cwaViewModelsAssisted(
+        factoryProducer = { viewModelFactory },
+        constructorCall = { factory, _ ->
+            factory as TraceLocationCreateViewModel.Factory
+            factory.create(navArgs.category)
+        }
+    )
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.toolbar.setNavigationOnClickListener {
+            it.hideKeyboard()
+            popBackStack()
+        }
+
+        viewModel.uiState.observe(viewLifecycleOwner) { state ->
+            binding.toolbar.setSubtitle(state.title)
+            binding.valueStart.text = state.getStartDate(requireContext().getLocale())
+            binding.valueEnd.text = state.getEndDate(requireContext().getLocale())
+            binding.layoutBegin.visibility = if (state.isDateVisible) View.VISIBLE else View.GONE
+            binding.layoutEnd.visibility = if (state.isDateVisible) View.VISIBLE else View.GONE
```

Comment:
`isVisible =  state.isDateVisible` ? and layoutBegin and layoutEnd have the same condition to show or hide , maybe a Constraint layout group would be helpful here ?
`layoutGroup.isVisible = state.isDateVisible`

Reply:
In this case, the parent layout is `LinearLayout`.
Refactoring to `Constraint Layout` would save 1 line of code. Let me know if you think if it worth the effort.

Reply:
I thought it is already a constraint , if not it is not worth it